### DISCUSSION TITLE
Scheduling Jobs missing assembly

### DISCUSSION THREAD

**Discussion by vhugogarcia** [shinyorg/shiny#748] [Q&A] [answered]
Hello friends,

I'm facing an issue when trying to schedule a job on Xamarin.Forms. On my viewmodel, I'm trying to schedule it by using the following code sample:
```
var job = new JobInfo
{
    Name = "YourFirstJob",
    Type = typeof(YourFirstJob),
    BatteryNotLow = true,
    DeviceCharging = false
    RequiredInternetAccess = InternetAccess.Any,
    Repeat = true //defaults to true, set to false to run once OR set it inside a job to cancel further execution
};
job.SetValue("Id", 10);
await ShinyHost.Resolve<Shiny.Jobs.IJobManager>().Schedule(job);
```

However on the line `await ShinyHost.Resolve<Shiny.Jobs.IJobManager>().Schedule(job);` I'm getting the following error: message:
**'IJobManager' does not contain a definition for 'Schedule' and no accessible method extension.**

I have the following packages installed on the project:
```
<PackageReference Include="Shiny.Core" Version="2.2.0.2829" />
    <PackageReference Include="Shiny.Notifications" Version="2.2.0.2829" />
    <PackageReference Include="Shiny" Version="2.2.0.2829" />
```

and the viewmodel class has the following:
```
using Shiny;
using Shiny.Jobs;
```

Am I missing something?

Thanks in advance

**Comment by aritchie** [comment, marked as answer]
The signature was changed in 2.x to Register

**Reply by vhugogarcia** [reply]
Thanks @aritchie for the clarification and help.  :)
